src/index.js

```diff
@@ -3,247 +3,267 @@
   Author Tobias Koppers @sokra
 */
 
-// TODO: temporarily disabled rules
-/* eslint-disable
-  no-undefined,
-  no-param-reassign,
-  no-underscore-dangle,
-  import/order
-*/
 import { SourceMapConsumer } from 'source-map';
-import { ConcatSource, RawSource, SourceMapSource } from 'webpack-sources';
-
+import { SourceMapSource, RawSource, ConcatSource } from 'webpack-sources';
 import RequestShortener from 'webpack/lib/RequestShortener';
 import ModuleFilenameHelpers from 'webpack/lib/ModuleFilenameHelpers';
+import uglify from 'uglify-es';
 
-import uglify from 'uglify-js';
+const defaultUglifyOptions = {
+  output: {
+    comments: /^\**!|@preserve|@license|@cc_on/,
+    beautify: false,
+    semicolons: true,
+    shebang: true,
+  },
+};
 
 class UglifyJsPlugin {
   constructor(options) {
     if (typeof options !== 'object' || Array.isArray(options)) {
-      options = {};
+      this.options = {};
+    } else {
+      this.options = options || {};
     }
-    if (typeof options.compressor !== 'undefined') {
-      options.compress = options.compressor;
+
+    this.options.test = this.options.test || /\.js($|\?)/i;
+    this.options.warningsFilter = this.options.warningsFilter || (() => true);
+
+    this.uglifyOptions = this.options.uglifyOptions || {};
+  }
+
+  static buildDefaultUglifyOptions({ ecma, warnings, parse, compress, mangle, output, toplevel, ie8 }) {
+    return {
+      ecma,
+      warnings,
+      parse: parse || {},
+      compress: compress || {},
+      // eslint-disable-next-line no-undefined
+      mangle: mangle === undefined || mangle === null ? true : mangle,
```

Comment:
@graingert I don't understand this. What you mean is that I should not test `mangle === null` but only ` mangle === undefined ? true : mangle` ?